nlmod/read/regis.py

```diff
@@ -163,18 +164,18 @@ def get_regis(
         msg = "No data found. Please supply valid extent in the Netherlands in RD-coordinates"
         raise (ValueError(msg))
 
-    # make sure layer names are regular strings
-    ds["layer"] = ds["layer"].astype(str)
+    regis_version = extract_version_from_title(ds.attrs["title"])
+
+    if rename_layers_to_version_2_2_2 and regis_version >= parse_version("2.2.3"):
+        pre223_layer_names = get_layer_names(rename_layers_to_version_2_2_2=True)
+        ds = ds.assign_coords({"layer": pre223_layer_names})
```

Comment:
This downloads regis a second time, and can be tricky when ds is changed before this line (in future commits). Maybe we can pass `ds` to `get_layer_names` as an optional parameter. We can then just do:
`ds["layer"] = get_layer_names(ds=ds, rename_layers_to_version_2_2_2=rename_layers_to_version_2_2_2)`
and remove some lines of double code, which is then only in `get_layer_names`.

Reply:
Thanks, great suggestion!